packages/core/strapi/lib/core-api/controller/index.d.ts

```diff
@@ -1,6 +1,6 @@
 import { Context } from 'koa';
 
-type Response = object;
+type Response <T=unknown> = T;
```

Comment:
```suggestion
type ControllerResponse <T=unknown> = T | Promise<T>;
```

This way, you can replace the SingleTypeController & CollectionTypeController response types with this shorthand. WDYT?